fission-cli/library/Fission/CLI/Handler/App/Publish.hs

```diff
@@ -103,17 +109,40 @@ publish watchFlag runner appURL appPath _updateDNS updateData = do -- TODO updat
               CLI.Error.put err "Server unable to sync data"
               raise err
 
-            Right _ ->
-              case watchFlag of
-                WatchFlag False ->
-                  success appURL
+            Right _ -> do
+              BaseUrl {..} <- getRemoteBaseUrl
+
+              let remoteUrl =
+                    (case baseUrlScheme of
+                        Https -> "https://"
+                        Http  -> "http://"
+                    )
+                    <> "ipfs."
+                    <> baseUrlHost
+                    <> "/ipns/"
+                    -- <> baseUrlPort -- not sure the best way to encorporate the port or if it's needed?
+
+              let open =
+                    liftIO . openBrowser $ remoteUrl <> show appURL
+
+              let watch =
+                    liftIO $ FS.withManager \watchMgr -> do
+                      hashCache <- newMVar hash
+                      timeCache <- newMVar =<< getCurrentTime
+                      void $ handleTreeChanges runner proof appURL updateData timeCache hashCache watchMgr absBuildPath
+                      forever . liftIO $ threadDelay 1_000_000 -- Sleep main thread
 
-                WatchFlag True ->
-                  liftIO $ FS.withManager \watchMgr -> do
-                    hashCache <- newMVar hash
-                    timeCache <- newMVar =<< getCurrentTime
-                    void $ handleTreeChanges runner proof appURL updateData timeCache hashCache watchMgr absBuildPath
-                    forever . liftIO $ threadDelay 1_000_000 -- Sleep main thread
+              case (openFlag, watchFlag) of
+                (OpenFlag True, WatchFlag True) ->
+                  open >> watch
+
+                (OpenFlag True, _) ->
+                  open >> success appURL
+
+                (_, WatchFlag True) -> watch
+
+                (OpenFlag False, WatchFlag False) ->
+                  success appURL
```

Comment:
No need for another mutable variable — awesome 🥳 

A couple Fission style things:

You don't need a separate `let` per declaration. We use `let` blocks with a newline:

```haskell
let 
  foo = 1
  bar = 2
  baz = 3
```

We avoid `>>`, since it's confusing for newcomers and the Elm folks 😛 Prefer `do` for consistency.

Instead of the `case` matrix, you could use `when`:

```haskell
publish (OpenFlag isOpen) (WatchFlag isWatched) runner appURL appPath _updateDNS updateData = do
  -- ... 

  when isOpen open

  when isWatched do
    -- put watcher code here instead of in the `let`

  success appURL
```

---

Ah, I see that we still have some of those pesky `MVars` kicking around 😱  I'll swap those over separately 👍